rskj-core/src/main/java/co/rsk/peg/federation/FederationSupport.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.peg.federation;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.UTXO;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.peg.vote.ABICallSpec;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+public interface FederationSupport {
+
+    Federation getActiveFederation();
+    Optional<Script> getActiveFederationRedeemScript();
+    Address getActiveFederationAddress();
+    int getActiveFederationSize();
+    Integer getActiveFederationThreshold();
+    Instant getActiveFederationCreationTime();
+    long getActiveFederationCreationBlockNumber();
+    byte[] getActiveFederatorBtcPublicKey(int index);
+    byte[] getActiveFederatorPublicKeyOfType(int index, FederationMember.KeyType keyType);
+    List<UTXO> getActiveFederationBtcUTXOs();
+
+    @Nullable
+    Federation getRetiringFederation();
```

Comment:
For another task, but what do you think about changing these functions to return an Optional instead of null?

Reply:
I like the idea. I'll add it to the backlog